cas-client-core/src/main/java/org/jasig/cas/client/util/URIBuilder.java

```diff
@@ -112,13 +112,14 @@ public URIBuilder(final URI uri) {
             final Charset utf8 = Charset.forName("UTF-8");
             if (query != null && !query.isEmpty()) {
                 final List<BasicNameValuePair> list = new ArrayList<BasicNameValuePair>();
-                final String queryValue = URLDecoder.decode(query, utf8.name());
-                final String[] parametersArray = queryValue.split("&");
+                final String[] parametersArray = query.split("&");
 
                 for (final String parameter : parametersArray) {
                     final String[] parameterCombo = parameter.split("=");
-                    if (parameterCombo.length == 2) {
-                        list.add(new BasicNameValuePair(parameterCombo[0], parameterCombo[1]));
+                    if (parameterCombo.length >= 1) {
+                        final String key = URLDecoder.decode(parameterCombo[0], utf8.name());
+                        final String val = parameterCombo.length == 2 ? URLDecoder.decode(parameterCombo[1], utf8.name()) : "";
```

Comment:
@mmoayyed thanks for review.
I think web application(CAS client) take different behavior when parameter key is not passed and when parameter has no value.
so, i think web filter should better not remove parameter key.